compiler/rustc_typeck/src/check/op.rs

```diff
@@ -673,6 +675,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     match actual.kind() {
                         Uint(_) if op == hir::UnOp::UnNeg => {
                             err.note("unsigned values cannot be negated");
+
+                            if let hir::ExprKind::Unary(
+                                _,
+                                hir::Expr {
+                                    kind:
+                                        hir::ExprKind::Lit(Spanned {
+                                            node: ast::LitKind::Int(1, _),
+                                            ..
+                                        }),
+                                    ..
+                                },
+                            ) = ex.kind
+                            {
+                                err.span_suggestion(
+                                    ex.span,
+                                    &format!("obtain the maximum `{}` value with its `MAX` associated constant", actual),
```

Comment:
Not sure if "its" is the right word here. "the" might work better.

Reply:
I think "its" is the right word here, since you want to talk about the possession of a "it".